crates/aiken-lang/src/expr.rs

```diff
@@ -1088,9 +1096,11 @@ impl UntypedExpr {
         generics: &mut IndexMap<u64, Rc<Type>>,
         data_types: &IndexMap<&DataTypeKey, &TypedDataType>,
         data: PlutusData,
-        tipo: &Type,
+        tipo: Rc<Type>,
     ) -> Result<Self, String> {
-        if let Type::App { name, module, .. } = tipo {
+        let tipo = Type::collapse_links(tipo);
```

Comment:
The essence of the change/fix is here ☝️, but had to rework a bit how `tipo` is passed and threaded through the reify function to keep the `Rc` around for convenience.